Port REP-2000 to ROS 2 documentation (#6249)
ros2/ros2_documentation: source/The-ROS2-Project/Release-Schedule.rst

Release Schedule
================

Frequency
---------

> New ROS 2 releases will be published in a time based fashion **every 12 months**.

Reviewer: ```suggestion
New ROS 2 releases are produced every **every 12 months**.
```

The rationale is that a shorter cycle (like 6 months) results in significant overhead and potentially many active releases at the same time (assuming they have the same support length).
On the other hand a longer cycle (like 2 years) is too long for users to wait for new features to be available in a ROS 2 release.

Targeted Platforms
------------------

> Since regular Ubuntu releases are only supported for 9 months ROS 2 will not target those.

Reviewer: ```suggestion
Since non-LTS (long term support) Ubuntu releases are only supported for 9 months, ROS 2 will not target those non-LTS Ubuntu releases.
```

A single ROS 2 distribution will only have full Tier 1 support for a **single** Ubuntu LTS.

> The rationale is that fully supporting two LTS versions - which means 2-year-different versions of upstream dependencies - is a tremendous overhead and sometimes even impossible.

Reviewer: ```suggestion
The rationale is that fully supporting two Ubuntu LTS versions is a tremendous overhead for our maintainers as there may be upstream dependencies that differ by up to two years .
```

On a case-by-case basis, a ROS 2 distribution may support an older Ubuntu LTS distribution as a Tier 3, community-supported platform.

> Since macOS (or at least brew) as well as Windows are rolling platforms we aim to support the latest version available at the time of a ROS 2 distribution release.

Reviewer: ```suggestion
Since both macOS (or at least brew) and Windows are rolling platforms, we aim to support the latest version available at the time of a ROS 2 distribution's release.
```

> For Debian we also aim to target the latest stable version though if that version is two years behind the Ubuntu version that might not be possible.

Reviewer: ```suggestion
For Debian we also aim to target the latest stable version; however, if that version is two years behind the Ubuntu version then it might not be possible.
```

Support
-------

LTS releases
^^^^^^^^^^^^

> Since Ubuntu LTS releases come with **5 years** of standard support we aim to match the same duration.

Reviewer: ```suggestion
Since Ubuntu LTS releases come with **5 years** of standard support, we aim for each ROS LTS distribution to have a similar support lifetime.
```

In even years new ROS 2 releases will happen one month after the Ubuntu LTS has been released (which usually means a ROS 2 release in May).

> The ROS 2 release will be supported until the end of the standard support of the Ubuntu LTS release which is 4 years and 11 months from the ROS 2 release date.

Reviewer: ```suggestion
The ROS 2 release will be supported until the end of the standard support window of the Ubuntu LTS release, which is 4 years and 11 months from the ROS 2 release date.
```
